clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,251 @@
+`master` branch into it. Resolve merge conflicts and test failures if any.
+```        
+    git checkout -b release_12.x master
+    OR
+    git checkout release_12.x
     git merge master
-
-You will very likely have merge conflicts and some test failures.  The test
-failures usually stem from incorrect merges or Checked C-specific data not being
-initialized by new or changed constructor methods.
-
-You may also need to pick up changes from LLVM/Clang for fixes to any unexpected
-baseline failures.
-
-You can push your updated master branches up to GitHub for automated
-testing.  If you haven't cleaned the build directory as described earlier,
-make sure you do that.
-
-You'll want to run automated tests on Linux and Windows x86/x64, as well as
+```     
+2. Upgrade the branch `release_12.x` to the LLVM/Clang sources corresponding to
+the 12.0.0 release. The commit hash associated with the 12.0.0 release is given
+by the LLVM/Clang tag `llvmorg-12.0.0`. Resolve merge conflicts and test
+failures if any. Record the fact that the `release_12.x` branch has been
+upgraded upto the commit hash given by the `llvmorg-12.0.0` tag, in the document
+LLVM-Upgrade-Notes.md.
+```
+    git remote add upstream https://github.com/llvm/llvm-project.git
+    git pull upstream llvmorg-12.0.0
+```
+3. Push the branch `release_12.x` to the remote repository, run ADO tests
+on it and fix test failures to make sure that the branch is stable.
+```
+    git push origin release_12.x
+```
+## Practical Tips
+
+1. If you use automated testing, make sure to clean the build directory first.
+Enough changes will likely have accumulated that things may go wrong without
+doing that first.
+
+2. The test failures usually stem from incorrect merges or Checked C-specific data
+not being initialized by new or changed constructor methods.
+
+3. You'll want to run automated tests on Linux and Windows x86/x64, as well as
 Linux LNT tests.  You may find in some cases that tests need to be updated
 based on new warnings or slightly different compiler behavior.
 
-## Merge your branched baseline and master branches
-
-Once all testing is passing, you can merge your branches back into
-your baseline and master branches.
-
-
-    git checkout baseline
-    git merge updated-baseline
-    git checkout master
-    git merge updated-master
-
-## Push the updated branches to GitHub
-
-The changes will be extensive enough that you don't want to do a pull request
+4. The changes will be extensive enough that you don't want to do a pull request
 on GitHub.  Just push the branches up to GitHub.
 
+
+## Handling Merge Conflicts and Test Failures
+
+1. When merge.renamelimit is set to a large value, the output of the `git merge`
+command may have several lines like:
+CONFLICT (rename/delete): ... in master renamed to ... in HEAD. Version HEAD ...
+left in tree.
+CONFLICT (rename/delete): ... in master renamed to ... in HEAD. Version master
+... left in tree.
```

Comment:
Format this as a code block rather than letting the two lines run together in the paragraph?